File: app/src/main/java/info/maaskant/wmsnotes/detail/DetailPagerAdapter.java
package info.maaskant.wmsnotes.detail;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

class DetailPagerAdapter extends FragmentPagerAdapter {

    private final EditorFragment editorFragment;
    private final ViewerFragment viewerFragment;

    DetailPagerAdapter(FragmentManager fragmentManager, EditorFragment editorFragment,
            ViewerFragment viewerFragment) {
        super(fragmentManager);
        this.editorFragment = editorFragment;
        this.viewerFragment = viewerFragment;
    }

    @Override
    public Fragment getItem(int position) {
        if (isEditorPage(position)) {
            return editorFragment;
        } else {
            return viewerFragment;
        }
    }

    @Override
    public int getCount() {
        return 2;
    }

    boolean isEditorPage(int position) {
        return position == 1;
    }

    boolean isViewerPage(int position) {
        return position != 1;
    }

}

File: app/src/main/java/info/maaskant/wmsnotes/android/ui/detail/DetailPagerAdapter.kt
package info.maaskant.wmsnotes.android.ui.detail

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

internal class DetailPagerAdapter(
    fragmentManager: FragmentManager,
    private val editorFragment: EditorFragment,
    private val viewerFragment: ViewerFragment
) : FragmentPagerAdapter(fragmentManager) {

    override fun getItem(position: Int): Fragment {
        return if (isEditorPage(position)) {
            editorFragment
        } else {
            viewerFragment
        }
    }

    override fun getCount(): Int = 2
    fun isEditorPage(position: Int): Boolean = position == 1
    fun isViewerPage(position: Int): Boolean = position != 1
}
